data_structures/src/chain.rs

```diff
@@ -733,6 +787,21 @@ pub struct PublicKey {
     pub bytes: [u8; 32],
 }
 
+/// Public Key data structure
+#[derive(Debug, Default, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct SecretKey {
+    pub bytes: [u8; 32],
```

Comment:
We should use `protected::Protected` type here, an array is a `Copy` type which is kind of spooky because the key will be copied all over the stack implicitly

Reply:
Created issue #560